e2e/test/scenarios/metabot/metabot.cy.spec.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  assertChatVisibility,
+  closeMetabotViaCloseButton,
+  closeMetabotViaShortcutKey,
+  describeEE,
+  metabotChatInput,
+  mockMetabotResponse,
+  openMetabotViaNewMenu,
+  openMetabotViaShortcutKey,
+  popover,
+  restore,
+  sendMetabotMessage,
+  setTokenFeatures,
+  userMessages,
+} from "e2e/support/helpers";
+
+describe("Metabot UI", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+    cy.intercept("POST", "/api/ee/metabot-v3/agent").as("agentReq");
+    cy.intercept("GET", "/api/session/properties").as("sessionProperties");
+  });
+
+  describe("OSS", { tags: "@OSS" }, () => {
+    beforeEach(() => {
+      cy.visit("/");
+      cy.wait("@sessionProperties");
+    });
+
+    it("should not be avaiable in OSS", () => {
+      openMetabotViaShortcutKey(false);
+      assertChatVisibility("not.visible");
+      cy.findByLabelText("Navigation bar").within(() => {
+        cy.findByText("New").click();
+      });
+      popover().findByText("Metabot request").should("not.exist");
+      assertChatVisibility("not.visible");
+    });
+  });
+
+  describeEE("EE", () => {
+    beforeEach(() => {
+      setTokenFeatures("all");
+      cy.visit("/");
+      cy.wait("@sessionProperties");
+    });
+
+    it("should be able to be opened and closed", () => {
+      openMetabotViaShortcutKey();
+      closeMetabotViaShortcutKey();
+      openMetabotViaNewMenu();
+      closeMetabotViaCloseButton();
+    });
+
+    it("should allow a user to send a message to the agent and handle successful or failed responses", () => {
+      openMetabotViaShortcutKey();
+      userMessages().should("not.exist");
+
+      mockMetabotResponse({
+        statusCode: 200,
+        delay: 100, // small delay to detect loading state
+        body: whoIsYourFavoriteResponse,
+      });
```

Comment:
This is interesting, It makes sense for testing the UI in a deterministic way, I wonder have you read up on what is the nascent state of the art for testing non-deterministic LLM responses?

Reply:
I am getting up to speed in this space, but I think collectively we're not yet sure about some core design decisions which greatly impact how we should test things. John and I are about to explore changing the API significantly, so once these things settle that's the next natural problem to solve.